ulib/libax/src/cbindings/mod.rs

```diff
@@ -38,27 +43,52 @@ pub unsafe extern "C" fn ax_rand_u32() -> u32 {
     crate::rand::rand_u32()
 }
 
+use spinlock::SpinNoIrq;
+// Lock used by ax_println_str for C apps
+static LOCK: SpinNoIrq<()> = SpinNoIrq::new(());
+
 /// Print a string to the global standard output stream.
 #[no_mangle]
 pub unsafe extern "C" fn ax_print_str(buf: *const c_char, count: usize) -> c_int {
     if buf.is_null() {
         return -axerrno::LinuxError::EFAULT.code();
     }
+
```

Comment:
Lock at here?

Reply:
LOCK seems to be unnecessary here, because next line `crate::io::stdout().write(bytes).unwrap() as _` uses lock.